ApollosProject/apollos-prototype
# Exposes getting / updating current campus

## DESCRIPTION

### What does this PR do, or why is it needed?

This PR adds two main parts to the schema, and removes a third part. 

1. Adds a field person to get their current campus
2. Adds a mutation to set a user's current campus (using a campus id)
3. Removes the family part of the schema. This was only used to get the families location, and now we are handling this using the campus schema exclusively. 

### What design trade-offs/decisions were made?

Decided to remove the campus schema. It wasn't needed for anything but the families location. We would rather not expose that, and would rather expose the user's campus.  

### How do I test this PR?

1. Login as a user in GraphIQL, set your auth token.
2. Check your current campus with this query
```
query getCurrentUser {
  currentUser {
    profile {
      birthDate
      campus {
        id
        name
      }
    }
  }
}
```
3. Get a list of campuses
```
query campuses {
  campuses {
    id
    name
  }
}
```
4. Update your campus.
```
mutation updateCampus {
  updateUserCampus(campusId:"Campus:a0f64573eabf00a607bec911794d50fb") {
    campus {
      id
      name
    }
  }
}
````
5. Check your current campus. Both in GraphIQL, and in the location header in the app. 

### What automated tests did you write?

100% coverage via resolver tests.

## TODO:

- [x] Closes #577 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
